src/sqlparser/src/ast/utils.rs

```diff
@@ -0,0 +1,574 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::fmt::{Debug, Display, Formatter};
+
+use serde::ser::{Impossible, StdError};
+use serde::{ser, Serialize};
+
+use crate::ast::{Ident, ObjectName, SqlOption, Value};
+
+#[derive(Debug)]
+pub enum Error {
+    Message(String),
+    NotSupported(String),
+}
+
+impl ser::Error for Error {
+    fn custom<T>(msg: T) -> Self
+    where
+        T: Display,
+    {
+        Error::Message(msg.to_string())
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Error::Message(msg) => f.write_str(msg),
+            Error::NotSupported(msg) => f.write_str(msg),
+        }
+    }
+}
+
+impl StdError for Error {}
+
+#[derive(Default)]
+struct ValueSerializer {}
+
+impl serde::Serializer for ValueSerializer {
+    type Error = Error;
+    type Ok = Value;
+    type SerializeMap = Impossible<Value, Error>;
+    type SerializeSeq = Impossible<Value, Error>;
+    type SerializeStruct = Impossible<Value, Error>;
+    type SerializeStructVariant = Impossible<Value, Error>;
+    type SerializeTuple = Impossible<Value, Error>;
+    type SerializeTupleStruct = Impossible<Value, Error>;
+    type SerializeTupleVariant = Impossible<Value, Error>;
+
+    fn serialize_bool(self, v: bool) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Boolean(v))
+    }
+
+    fn serialize_i8(self, v: i8) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_i16(self, v: i16) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_i32(self, v: i32) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_i64(self, v: i64) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_u8(self, v: u8) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_u16(self, v: u16) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_u32(self, v: u32) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_u64(self, v: u64) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_f32(self, v: f32) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_f64(self, v: f64) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::Number(v.to_string()))
+    }
+
+    fn serialize_char(self, v: char) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::SingleQuotedString(v.to_string()))
+    }
+
+    fn serialize_str(self, v: &str) -> Result<Self::Ok, Self::Error> {
+        Ok(Value::SingleQuotedString(v.to_string()))
```

Comment:
If there's a special character (like `\t`), I guess we should serialize into a `CstyleEscapedString` instead. https://github.com/risingwavelabs/risingwave/pull/14193#issuecomment-1874816439